src/unix/cygwin/mod.rs

```diff
@@ -0,0 +1,3137 @@
+use crate::prelude::*;
+use crate::*;
+
+pub type c_char = i8;
+pub type c_long = i64;
+pub type c_ulong = u64;
+
+cfg_if! {
+    if #[cfg(doc)] {
+        pub(crate) type Ioctl = c_int;
+    } else {
+        #[doc(hidden)]
+        pub type Ioctl = c_int;
+    }
+}
+
+pub type wchar_t = c_ushort;
+
+pub type blkcnt_t = i64;
+pub type blksize_t = i32;
+pub type dev_t = u32;
+pub type fsblkcnt_t = c_ulong;
+pub type fsfilcnt_t = c_ulong;
+pub type ino_t = u64;
+pub type key_t = c_longlong;
+pub type sa_family_t = u16;
+pub type socklen_t = c_int;
+
+pub type off_t = c_long;
+pub type id_t = u32;
+pub type mode_t = u32;
+pub type _off64_t = c_longlong;
+pub type loff_t = _off64_t;
+pub type iconv_t = *mut c_void;
+pub type clock_t = c_ulong;
+pub type time_t = c_long;
+pub type clockid_t = c_ulong;
+pub type timer_t = c_ulong;
+pub type nl_item = c_int;
+pub type nlink_t = c_ushort;
+pub type suseconds_t = c_long;
+pub type useconds_t = c_ulong;
+
+#[cfg_attr(feature = "extra_traits", derive(Debug))]
+pub enum timezone {}
+impl Copy for timezone {}
+impl Clone for timezone {
+    fn clone(&self) -> timezone {
+        *self
+    }
+}
```

Comment:
Can this go in an `e!` macro so it gets these derived? Same for `enum sem`.

Reply:
The other targets use this way of definition.

Reply:
The macro `e!` also impls Eq, Hash, PartialEq. We don't want those traits.